src/ngAnimate/animateCss.js

```diff
@@ -588,17 +590,10 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
         blockTransitions(node, SAFE_FAST_FORWARD_DURATION_VALUE);
       }
 
-      var timings = computeTimings(node, fullClassName, cacheKey);
-      var relativeDelay = timings.maxDelay;
-      maxDelay = Math.max(relativeDelay, 0);
-      maxDuration = timings.maxDuration;
-
       var flags = {};
       flags.hasTransitions          = timings.transitionDuration > 0;
       flags.hasAnimations           = timings.animationDuration > 0;
-      flags.hasTransitionAll        = flags.hasTransitions && timings.transitionProperty == 'all';
-      flags.applyTransitionDuration = hasToStyles && (
-                                        (flags.hasTransitions && !flags.hasTransitionAll)
+      flags.applyTransitionDuration = options.duration > 0 || hasToStyles && (flags.hasTransitions
                                          || (flags.hasAnimations && !flags.hasTransitions));
```

Comment:
Yes, but what if the element has transition styles with `transition-duration > 0` whereas `options.duration === 0`.
E.g. the user wants to have a 0-duration animation (i.e. skip the animation for whatever reason), on an element that would normally animate (because it has `transition-duration` > 0).

Or is it so edgy a case, that we don't need to support it ? :smiley: